metagraph/CMakeListsHelpers.txt

```diff
@@ -72,44 +72,57 @@ endfunction()
 
 function (findFolly)
 unset(FOLLY_LIBRARIES)
-find_library(FOLLY_LIBRARIES
-  NAMES folly
+find_package(Boost)
+find_library(jemalloc_FOUND
+  NAMES jemalloc
   HINTS
   ENV LD_LIBRARY_PATH
   PATHS
     $ENV{HOME}/.linuxbrew/lib
 )
-if(FOLLY_LIBRARIES)
-  string(APPEND CMAKE_CXX_FLAGS " -D_USE_FOLLY")
-  find_package(OpenSSL REQUIRED COMPONENTS Crypto)
-  set(FOLLY_LIBRARIES ${FOLLY_LIBRARIES}
-    -ldouble-conversion
-    -ldl
-    -lboost_regex
-    -lboost_filesystem
-    -lboost_program_options
-    OpenSSL::Crypto
+
+if(!jemalloc_FOUND)
+  message(WARNING "jemalloc not found.")
+endif()
+
+find_library(tcmalloc_FOUND
+  NAMES tcmalloc
+  HINTS
+  ENV LD_LIBRARY_PATH
+  PATHS
+    $ENV{HOME}/.linuxbrew/lib
+)
+
+if(!tcmalloc_FOUND)
+  message(WARNING "gperftools not found")
+endif()
+
+if(jemalloc_FOUND AND tcmalloc_FOUND AND Boost_FOUND)
+message(STATUS "Configuring Folly")
+  string(APPEND CMAKE_CXX_FLAGS " -D_USE_FOLLY -DUSE_JEMALLOC -DUSE_TCMALLOC")
+  include(external-libraries/folly/CMake/FollyConfigChecks.cmake)
+  configure_file(
+    external-libraries/folly/CMake/folly-config.h.cmake
+    ${CMAKE_BINARY_DIR}/external-libraries/folly/include/folly/folly-config.h
```

Comment:
can we move all this this to another cmake file find_folly, similarly to CMakeListsKMC.txt.in?